diff -ruN a/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_dsp_core_3000.cpp b/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_dsp_core_3000.cpp
--- a/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_dsp_core_3000.cpp	2021-06-26 10:35:29.711689016 +0800
+++ b/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_dsp_core_3000.cpp	2021-06-26 10:37:17.232689124 +0800
@@ -12,6 +12,9 @@
 #include <uhd/utils/safe_call.hpp>
 #include <uhdlib/usrp/cores/dsp_core_utils.hpp>
 #include <uhdlib/usrp/cores/rx_dsp_core_3000.hpp>
+#include <boost/assign/list_of.hpp>
+#include <boost/math/special_functions/round.hpp>
+#include <boost/math/special_functions/sign.hpp>
 #include <algorithm>
 #include <cmath>
 #include <functional>
diff -ruN a/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_frontend_core_3000.cpp b/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_frontend_core_3000.cpp
--- a/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_frontend_core_3000.cpp	2021-06-26 10:35:29.711689016 +0800
+++ b/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/cores/rx_frontend_core_3000.cpp	2021-06-26 10:39:39.246689267 +0800
@@ -10,7 +10,9 @@
 #include <uhd/utils/math.hpp>
 #include <uhdlib/usrp/cores/dsp_core_utils.hpp>
 #include <uhdlib/usrp/cores/rx_frontend_core_3000.hpp>
-#include <cmath>
+#include <boost/assign/list_of.hpp>
+#include <boost/math/special_functions/round.hpp>
+#include <boost/math/special_functions/sign.hpp>
 #include <functional>
 
 using namespace uhd;
diff -ruN a/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/usrp2/usrp2_impl.cpp b/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/usrp2/usrp2_impl.cpp
--- a/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/usrp2/usrp2_impl.cpp	2021-06-26 10:35:29.714689016 +0800
+++ b/uhd-f3d3b071fd193bf8685c86e74f32b7b96b5e3be2/host/lib/usrp/usrp2/usrp2_impl.cpp	2021-06-26 10:39:58.888689287 +0800
@@ -17,8 +17,10 @@
 #include <uhdlib/usrp/common/apply_corrections.hpp>
 #include <boost/asio.hpp>
 #include <boost/asio/ip/address_v4.hpp>
+#include <boost/math/special_functions/round.hpp>
+#include <boost/math/special_functions/sign.hpp>
 #include <boost/format.hpp>
-#include <cmath>
+#include <boost/thread.hpp>
 #include <functional>
 
 using namespace uhd;
@@ -946,9 +948,6 @@
     return meta_range_t(-dac_rate / 2, +dac_rate / 2, dsp_range_step);
 }
 
-#include <boost/math/special_functions/round.hpp>
-#include <boost/math/special_functions/sign.hpp>
-
 void usrp2_impl::update_clock_source(const std::string& mb, const std::string& source)
 {
     // NOTICE: U2_REG_MISC_CTRL_CLOCK is on the wb clock, and cannot be set from fifo_ctrl
